### PR TITLE
Sax parser

### LINKED ISSUE #2: Streaming XML Parser
Some RETS servers (at least a couple Marketlinx boards) handle pagination badly, and so require seriously huge queries if you don't want to miss anything. This will blow up a delayed job using standard DOM `Nokogiri.parse`. Stream parsing, especially with a lazy `#each`, makes this less painful.

Check out https://github.com/josephholsten/rets4r/blob/master/lib/rets4r/client/parsers/compact_nokogiri.rb for a parser that returns a hash for each row. It's under the ruby license, so you're welcome to use it under your MIT.

### PR DESCRIPTION
**Pull request by dougcole**
Only for data fetches, not metadata or counts, but that covers the biggest results in my experience. Fixes #2

### CODE REVIEW

File: lib/rets/parser/compact.rb [hunk 1]
```diff
@@ -1,32 +1,61 @@
...
-        rows.map do |data|
-          self.parse(columns, data.text, delimiter)
+        def characters string
+          if @columns_start
+            @columns = string
+          end
+
+          if @data_start
+            @results << string
+            # hi doug
```
Comment (dougcole) [inline lib/rets/parser/compact.rb]: DAMNIT @phiggins!

**Comment by tdtran** [comment]
Code looks OK but I'd recommend to test it against a real RETS server before merging into master.

**Comment by dougcole** [comment]
Thanks. Turns out one of the pull I merged this weekend (https://github.com/estately/rets/pull/101) conflicts quite a bit with this, so it'll be a bit before I resolve all the conflicts, but I'll get this ready again soon.

**Comment by hfaulds** [comment]
Fixed up and should be ready to go. I did have to change the parser slightly to handle nested tags within the results field.

**Comment by hfaulds** [comment]
I have 0 idea why this would break on ruby 1.9.2 if any-one else wants to pick this up then feel free

**Comment by phiggins** [comment]
Not sure if this is the right way to fix it, but adding a magic comment got the build to pass: https://github.com/estately/rets/commit/24c7d54ee53d994256b64ae334aa06880338082a

**Comment by dougcole** [comment]
Seems like a reasonable fix. Let's finish testing https://github.com/estately/rets/pull/134 in production and then test/merge this next.

**Comment by hfaulds** [comment]
Going to start testing this in production now.

**Comment by hfaulds** [comment]
We had an unrelated issue in https://github.com/estately/rets/pull/147 but otherwise this seems to have worked fine as far as I can tell.

**Pull request merged**